torch/autograd/_functions/reduce.py

```diff
@@ -159,7 +159,7 @@ def forward(cls, ctx, input, dim=None, keepdim=True, additional_args=tuple()):
     def backward(cls, ctx, grad_output, grad_indices=None):
         grad_input = Variable(grad_output.data.new(*ctx.input_size).zero_())
         if ctx.dim is None and cls.has_all_reduce:
-            grad_input[ctx.indices_tuple] = grad_output.data[0]
+            grad_input[ctx.indices_tuple] = grad_output[0]
```

Comment:
It ends up calling Index.apply, right?

>>> y=Variable(torch.randn(5,6), requires_grad=True)
>>> y[0].grad_fn
<torch.autograd.function.IndexBackward object at 0x7fa2c45e9de0>